win32ss/user/ntuser/ime.c

```diff
@@ -80,14 +80,72 @@ NtUserCheckImeHotKey(
     return 0;
 }
 
-
-DWORD
+BOOL
 APIENTRY
 NtUserDisableThreadIme(
-    DWORD dwUnknown1)
+    DWORD dwThreadID)
 {
-    STUB;
-    return 0;
+    PTHREADINFO pti, ptiCurrent;
+    PPROCESSINFO ppi;
+    BOOL ret = FALSE;
+
+    UserEnterExclusive();
+
+    if (!IS_IMM_MODE())
+    {
+        EngSetLastError(ERROR_CALL_NOT_IMPLEMENTED);
+        goto Quit;
+    }
+
+    ptiCurrent = GetW32ThreadInfo();
+
+    if (dwThreadID == (DWORD)-1)
+    {
+        ppi = ptiCurrent->ppi;
+        ppi->W32PF_flags |= W32PF_DISABLEIME;
+
+        for (pti = ppi->ptiList; pti; )
+        {
+            pti->TIF_flags |= TIF_DISABLEIME;
+
+            if (pti->spwndDefaultIme)
+            {
+                co_UserDestroyWindow(pti->spwndDefaultIme);
+                pti->spwndDefaultIme = NULL;
+                pti = ptiCurrent->ppi->ptiList;
+                continue;
+            }
+
+            pti = pti->ptiSibling;
+        }
+    }
+    else
+    {
+        if (dwThreadID == 0)
+        {
+            pti = ptiCurrent;
+        }
+        else
+        {
+            pti = IntTID2PTI(UlongToHandle(dwThreadID));
+            if (!pti || pti->ppi != ptiCurrent->ppi)
```

Comment:
Add a comment that the thread needs to reside in the current process.

Reply:
Added.